lib/OpenQA/WebAPI/Controller/API/V1/User.pm

```diff
@@ -0,0 +1,53 @@
+package OpenQA::WebAPI::Controller::API::V1::User;
+use Mojo::Base 'Mojolicious::Controller';
+
+=pod
+
+=head1 NAME
+
+OpenQA::WebAPI::Controller::API::V1::User
+
+=head1 SYNOPSIS
+
+  use OpenQA::WebAPI::Controller::API::V1::User;
+
+=head1 DESCRIPTION
+
+Implements User API.
+
+=head1 METHODS
+
+=over 4
+
+=item delete()
+
+Post integer value to save User tour progress of current user in the database
+
+=back
+
+=cut
+
+sub delete {
+    my ($self)      = @_;
+    my $users       = $self->schema->resultset('Users');
+
+    if ($self->current_user->is_admin != 1){
+        return $self->render(json => {error => 'Only administrators could delete users'}, status => 401);
+    }
+
+    my $user = $users->find($self->param('id'));
+    if (!$user) {
+        return $self->render(json => {error => 'Not found'}, status => 404);
+    }
```

Comment:
I guess the preferred coding style (also by other team members, not just me) is `return $self->render(json => {error => 'Not found'}, status => 404) unless my $user = $users->find($self->param('id'));` and then no extra block for the else part. The same counts for the `!ret` case below.

Reply:
Wasn't the `unless my` something that can lead to bugs due to the conditional declaration?

Reply:
Previously, I leaved the block, because I had an two different blocks, but I agree to use unless. Take a look to the new code.

Reply:
done AFAICS